src/vs/base/node/id.ts

```diff
@@ -99,3 +100,16 @@ async function getMacMachineId(errorLogger: (error: any) => void): Promise<strin
 		return undefined;
 	}
 }
+
+const SQM_KEY: string = '\\Software\\Microsoft\\SQMClient';
+export async function getSqmMachineId(): Promise<string | undefined> {
+	if (isWindows) {
```

Comment:
I wonder if we need to be more resilient against issues and potentially slow lookup times:
- should we always return an empty `string` in error cases so that we only resolve this ID once and not on each startup?
- we should probably log errors to log service when resolving fails
- should we have a timeout after which we return and give up just in case this call blocks for a long time?

Reply:
First two points tackled. Third point - let's see if we can move it so it non blocking startup. If we see it blocks startup then we can introduce a timeout.